src/System.Slices/System/ReadOnlySpan.cs

```diff
@@ -338,5 +338,19 @@ public override bool Equals(object obj)
             }
             return false;
         }
+
+        public static bool operator ==(ReadOnlySpan<T> span1, ReadOnlySpan<T> span2) => span1.Equals(span2);
+        public static bool operator !=(ReadOnlySpan<T> span1, ReadOnlySpan<T> span2) => !span1.Equals(span2);
+        public bool Equals(Span<T> other) => other.ReferenceEquals(Object, Offset, Length);
```

Comment:
I'd consider calling it something other than `ReferenceEquals`.  That is really easy to confuse with `object.ReferenceEquals` and this is not actually doing reference equality checks under the hood.  It's doing a combination of reference and value equality.